winch/codegen/src/visitor.rs

```diff
@@ -117,6 +118,10 @@ where
         self.context.stack.push(Val::i32(val));
     }
 
+    fn visit_drop(&mut self) -> Self::Output {
+        self.context.drop_last(1);
```

Comment:
`drop_last` will free any registers but it won't free any memory if the top value is `Mem` entry. 

To fix this I see two options:

1. Before calling `drop_last` peek the top value and if it is a memory entry, emit a `masm.free_stack(ABI::word_bytes())`
2. Extend `drop_last` to free any memory entries.

The reason why I have designed this in such a way is that freeing each memory entry one-by-one involves emitting too many instructions when in reality it could be be done in a single instruction as long as you know the amount of bytes that you need to free.

One solution to extending `drop_last` would be to perhaps allow it to take a callback that will be invoked everytime a `mem` entry is found, that way the  caller can decide what to do with it, in the case of `drop` it will free the memory for that `Mem` entry:


```rust
drop_last(1, |val, masm| {
   masm.free_stack(...);
});
```

But doing option 1 is totally fine too, we don't have to get into deep refactoring to land this.